pkg/cloudprovider/provider/hetzner/provider.go

```diff
@@ -152,6 +161,14 @@ func (p *provider) Validate(spec v1alpha1.MachineSpec) error {
 		}
 	}
 
+	if len(c.Networks) != 0 {
+		for _, network := range c.Networks {
+			if _, _, err = client.Network.Get(ctx, network); err != nil {
+				return fmt.Errorf("failed to get network: %v", err)
```

Comment:
nit: Can we put the name of the network we couldnt get into the error?

Reply:
Fixed